infrastructure/comms/src/connection/net_address/net_addresses.rs

```diff
@@ -0,0 +1,253 @@
+use crate::connection::{
+    net_address::{net_address_with_stats::NetAddressWithStats, NetAddressError},
+    NetAddress,
+};
+use chrono::prelude::*;
+
+const MAX_CONNECTION_ATTEMPTS: u32 = 3;
+
+#[derive(Debug)]
+pub struct NetAddresses {
```

Comment:
Could you add a comment here indicating that this struct represents a list of different addresses (ipv4, tor, etc) for _a single peer_? This wasn't clear until deep into the code review and then suddenly some of the choices here made a lot more sense.

Reply:
Added a comment explaining the different address types handled by the struct.